lib/elixir/lib/kernel.ex

```diff
@@ -2924,6 +2926,10 @@ defmodule Kernel do
     end
   end
 
+  defp comp(left, {:|, _, [h, t]}) do
+    quote(do: :erlang.or(unquote(left) in unquote(t), :erlang."=:="(unquote(left), unquote(h))))
```

Comment:
Yeah, I wasn't sure about this but I went with the same order that other optimized in/2 cases use, such as this example from the tests:

```elixir
assert expand_to_string(quote(do: foo in [1, 2])) ==
             ~S[:erlang.or(:erlang.=:=(foo, 2), :erlang.=:=(foo, 1))]
```